### PR TITLE
Refactor package READMEs to link to shopify.dev docs

### PR DESCRIPTION
**Pull request by lesliexin**
### **Summary**

To ensure we have a single source of truth for the documentation of our packages, we should link to the shopify.dev docs in the READMEs. 

**Pros**
- Single source of truth 
- Easier to maintain 
- Follows Shopify conventions 

## 🕹️ Demonstration

- [Connect wallet](https://github.com/Shopify/blockchain-components/blob/docs/link-to-dot-dev-in-readme/packages/connect-wallet/README.md) 
- [Gate context client](https://github.com/Shopify/blockchain-components/blob/docs/link-to-dot-dev-in-readme/packages/gate-context-client/README.md)

## ✅ Checklist

- [x] ~~Tested on mobile~~
- [x] Tested on multiple browsers
- [x] ~~Tested for accessibility~~
- [x] ~~Includes unit tests~~
- [x] ~~Updated relevant documentation for the changes (if necessary)~~

### CODE REVIEW

**Review by caropinzonsilva — APPROVED** [review]
The code looks good! I would just wait for an answer from the team to understand how we would want to handle the README (related to the slack thread Q posted)

File: packages/gate-context-client/README.md [hunk 1]
```diff
@@ -4,7 +4,11 @@
 
 The `gate-context-client` package provides an abstract interface to read and write gate context information. The client writes gate context information for use by [Shopify functions](/docs/api/functions). For example, the client could write a signed message to the context. That message could be validated and processed by a Shopify function in order to apply discounts or block checkout.
 
-## Code example
+## Documentation
+
+See the [Tokengating example app tutorial](https://shopify.dev/apps/blockchain/tokengating/build-a-tokengating-app) for further documentation. Specifically, the end of the [Show gates on the storefront using a theme app extension](https://shopify.dev/apps/blockchain/tokengating/build-a-tokengating-app/show-gates-storefront) part about writing an HMAC.
+
+### Code example
```
[the remark this answers is not in the document]
Comment (jamiely) [reply packages/gate-context-client/README.md:11]: Sorry, no gate context-client doesn't have documentation in shopify.dev yet. There is only 1 card that links to the package on NPM

**Review by jamiely — APPROVED** [review]
:+1: I am good with this

**Comment by lesliexin** [comment]
Will update this PR with the following proposed structure:

- NPM links
- Description
- Link to documentation
- Installation
- Contributing
- Licenses


More context in this [slack](https://shopify.slack.com/archives/C03NL8QQ6DU/p1679343753338959?thread_ts=1678996881.140609&cid=C03NL8QQ6DU) thread

**Comment by lesliexin** [comment]
In Slack, we came to the conclusion to add an additional simple getting started section. 

I tried doing that with the suggestion of only including a code snippet for the `ConnectWalletProvider` and linking out to the shopify.dev docs for the other steps (1 and 3)

Screenshot 2023-03-20 at 6 30 59 PM

But looking at the screenshot above, I don't think it makes sense to include a code snippet in the README for only step 2 when we are forcing the devs to go to shopify.dev anyways for steps 1 and 3. In my mind it's either:
- link to shopify.dev and not include any code in the README
- or include **all** the code necessary for getting started in the README

But an in-between doesn't necessarily make sense. 

Therefore since we decided that we wanted a getting started section, I included the code snippets for all 3 steps as they are the bare minimum for getting started without needing to visit shopify.dev. 

If needed we can revisit if we get feedback or we find it's difficult to maintain, but for now let's move forward with this.

File: packages/gate-context-client/README.md [hunk 2]
```diff
@@ -26,17 +30,13 @@ async function run(data: Input) {
...
 ```
 
-## Installation
+### Installation
 
 ```bash
 yarn add @shopify/gate-context-client
 ```
 
-## Get started
```
[the remark this answers is not in the document]
Comment (lesliexin) [reply packages/gate-context-client/README.md:35]: The same content is still here just moved to line 7 to align with the same structure as the other READMEs

File: packages/gate-context-client/README.md [hunk 3]
```diff
@@ -26,17 +30,13 @@ async function run(data: Input) {
 }
 ```
 
-## Installation
+### Installation
```
[the remark this answers is not in the document]
Comment (lesliexin) [reply packages/gate-context-client/README.md:33]: Ooh good catch I'll fix that right now!

**Pull request merged**